building/build-debs/homeworld-keyserver/src/keycommon/knc/knc.go

```diff
@@ -0,0 +1,27 @@
+package knc
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+func Request(host string, data []byte) ([]byte, error) {
+	cmd := exec.Command("/usr/bin/knc", fmt.Sprintf("host@%s", host), "20575")
+
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		defer stdin.Close()
+		stdin.Write(data)
+	}()
+
+	response, err := cmd.Output()
```

Comment:
How do timeouts and other errors show up in this output? Are they converted to errors?

Reply:
According to the golang docs, if the command exits unsuccessfully, Cmd.Output() will return an ExitError containing a snippet of stderr.

Reply:
Can you confirm that this is the case?

Reply:
Yup, confirmed, the returned error is non-nil and a snippet of stderr is stored in `err.(*exec.ExitError).Stderr` as a bytestring.